app/qml/inputs/MMTextAreaEditor.qml

```diff
@@ -0,0 +1,47 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+import QtQuick
+import QtQuick.Controls
+import QtQuick.Controls.Basic
+import QtQuick.Layouts
+import Qt5Compat.GraphicalEffects
+import "../components"
+
+MMAbstractEditor {
+  id: root
+
+  property var parentValue: parent.value ?? ""
+  property bool parentValueIsNull: parent.valueIsNull ?? false
+  property bool isReadOnly: parent.readOnly ?? false
+
+  property alias placeholderText: textArea.placeholderText
+  property alias text: textArea.text
+
+  signal editorValueChanged( var newValue, var isNull )
+
+  hasFocus: textArea.activeFocus
+  innerHeight: textArea.vertSpace + textArea.contentHeight + 2 * textArea.vertSpace
+
+  content: TextArea {
+    id: textArea
+
+    property real vertSpace: 11 * __dp
+
+    y: textArea.vertSpace
```

Comment:
Why do you need to set `y` here?

Reply:
Because of pixel-perfect design, resizable content needs more adjustments.